### PR TITLE
use Predicate for predicates #3030

### PR DESCRIPTION
**Pull request by lindner**

### CODE REVIEW

File: src/runtime/slot-consumer.ts [hunk 1]
```diff
@@ -81,7 +82,7 @@ export class SlotConsumer {
     return [...this.generateProvidedContexts()];
   }
 
-  findProvidedContext(predicate: (_: ProvidedSlotContext) => boolean) {
+  findProvidedContext(predicate: Predicate<ProvidedSlotContext>) {
```
Comment (mmandlis) [inline src/runtime/slot-consumer.ts]: did you skip the one on line 89 on purpose?

File: src/devtools-connector/arc-planner-invoker.ts [hunk 2]
```diff
@@ -191,13 +192,13 @@ export class ArcPlannerInvoker {
     return {suggestion, error: ((({location, message}) => ({location, message}))(error))};
   }
 
-  findManifestNames(manifest: Manifest, predicate: (_ : Manifest) => boolean): string[] {
+  findManifestNames(manifest: Manifest, predicate: Predicate<Manifest>): string[] {
```
Comment (alxrsngrtn) [inline src/devtools-connector/arc-planner-invoker.ts]: much cleaner. 👍

File: src/runtime/interface-info.ts [hunk 3]
```diff
@@ -10,6 +10,7 @@
 
 import {assert} from '../platform/assert-web.js';
 
+import {Predicate} from '../runtime/hot.js';
 import {TypeChecker} from './recipe/type-checker.js';
 import {Type, TypeVariable, TypeLiteral} from './type.js';
```
Comment (alxrsngrtn) [inline src/runtime/interface-info.ts]: Should imports be alphabetical?

File: src/runtime/slot-consumer.ts [hunk 4]
```diff
@@ -81,7 +82,7 @@ export class SlotConsumer {
     return [...this.generateProvidedContexts()];
```
Comment (alxrsngrtn) [inline src/runtime/slot-consumer.ts]: 47, 48: The callbacks can be consumers.

Comment (alxrsngrtn) [reply src/runtime/slot-consumer.ts]: +49

File: src/runtime/slot-consumer.ts (hunk 1 above)
Comment (lindner) [reply src/runtime/slot-consumer.ts]: Thanks!  I looked for the regex `=>.boolean`, but this was a specific usage. (=> true)

Fixed.

File: src/runtime/interface-info.ts (hunk 3 above)
Comment (lindner) [reply src/runtime/interface-info.ts]: there was a lint fixer but we encountered errors.  Will likely revisit as we move from tslint -> eslint.

File: src/runtime/slot-consumer.ts (hunk 4 above)
Comment (lindner) [reply src/runtime/slot-consumer.ts]: Next PR :-P

Comment (alxrsngrtn) [reply src/runtime/slot-consumer.ts]: 👍

**Pull request merged**